### PR TITLE
Print to 0 dp if genome pos arrays are integers

### PR DESCRIPTION
**Pull request by hyanwong**
Fixes https://github.com/tskit-dev/tskit/issues/1145 - where should the tests go?

### CODE REVIEW

**Comment by jeromekelleher** [comment]
LGTM - where are we testing this text display stuff @benjeffery?

**Comment by benjeffery** [comment]
We don't test the exact content of the text output, there is only the very simple `test_str` and `test_repr_html`.
You could make `test_str` more meaningful if you like?

**Comment by hyanwong** [comment]
I'll add something. For the DTWF I wonder if we should also print time to 0 dp if all times are integers?

**Comment by hyanwong** [comment]
New test add, which is a bit fiddly, since (a) we need to extract the str representation from the table string, which I hack by splitting on "│", and (b) tables with `left` and `right` only show integers when both left and right columns are entirely integer. But it works!

File: python/tskit/tables.py [hunk 1]
```diff
@@ -1088,6 +1088,9 @@ def __init__(self, max_rows_increment=0, ll_table=None):
             ll_table = _tskit.NodeTable(max_rows_increment=max_rows_increment)
         super().__init__(ll_table, NodeTableRow)
 
+    def _has_discrete_times(self):
```
Comment (benjeffery) [inline python/tskit/tables.py:1091]: This is going to sound picky, but I'm not a fan of one-use functions like this. If we could see ourselves using this a lot then I think it's worth it, but I don't see us reusing these so it isn't worth breaking up the flow and adding to the large list of methods on a Table.

File: python/tskit/tables.py [hunk 2]
```diff
@@ -1717,6 +1741,9 @@ def __init__(self, max_rows_increment=0, ll_table=None):
             ll_table = _tskit.SiteTable(max_rows_increment=max_rows_increment)
         super().__init__(ll_table, SiteTableRow)
 
+    def _has_discrete_coords(self):
+        return np.all(self.position) == np.floor(self.position)
```
Comment (benjeffery) [inline python/tskit/tables.py:1745]: No coverage here as "position" needs to be in the tests.

**Review by benjeffery — APPROVED** [review]
Great, thanks @hyanwong. One nit about adding one-use functions.

File: python/tskit/tables.py (hunk 2 above)
Comment (hyanwong) [reply python/tskit/tables.py:1745]: Oh, of course. Doh! sorry.

File: python/tskit/tables.py (hunk 1 above)
Comment (hyanwong) [reply python/tskit/tables.py:1091]: I'm happy to go with whatever you suggest. I'm just following what @jeromekelleher suggested (that's me passing the buck)

Comment (benjeffery) [reply python/tskit/tables.py:1091]: Sure, that would be good.

Comment (hyanwong) [reply python/tskit/tables.py]: Done. Much tidier now, good call.

File: python/tskit/tables.py [hunk 3]
```diff
@@ -681,6 +681,13 @@ def _repr_html_(self):
             </div>
         """
 
+    def _columns_all_integer(self, *colnames):
+        """For displaying floating point values without loads of decimal places"""
```
[the remark this answers is not in the document]
Comment (hyanwong) [reply python/tskit/tables.py:685]: Ah, nice catch. I didn't realise the distinction.

Might be worth putting this in the tskit dev style guide? If it's not there already.

**Comment by hyanwong** [comment]
> One last nit!

At times I've wished we were at that stage with the kid's hair.

**Pull request merged**